pytorch_lightning/strategies/hpu_parallel.py

```diff
@@ -0,0 +1,145 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+import os
+from typing import Dict, List, Optional
+
+import torch
+import torch.distributed
+
+import pytorch_lightning as pl
+from pytorch_lightning.overrides import LightningDistributedModule
+from pytorch_lightning.overrides.torch_distributed import broadcast_object_list
+from pytorch_lightning.plugins.io.checkpoint_plugin import CheckpointIO
+from pytorch_lightning.plugins.io.hpu_plugin import HPUCheckpointIO
+from pytorch_lightning.plugins.precision import PrecisionPlugin
+from pytorch_lightning.strategies.ddp import DDPStrategy
+from pytorch_lightning.utilities.distributed import group as _group
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.imports import _HPU_AVAILABLE, _TORCH_LESSER_EQUAL_1_10_2
+from pytorch_lightning.utilities.rank_zero import rank_zero_warn
+
+if _HPU_AVAILABLE:
+    import habana_frameworks.torch.core.hccl  # noqa: F401
+    from habana_frameworks.torch.utils.library_loader import load_habana_module
+
+log = logging.getLogger(__name__)
+
+
+class HPUParallelStrategy(DDPStrategy):
+    """Strategy for distributed training on multiple HPU devices."""
+
+    # The main process in each node spawns N-1 child processes via :func:`subprocess.Popen`, where N is the number of
+    # devices (e.g. GPU) per node. It is very similar to how :mod:`torch.distributed.launch` launches processes.
+
+    # Multi-device per process is not supported with habana
```

Comment:
Do we need those comments? To me they just summarize how DDP works, so this explanations doesn't have to be doubled here  :)
```suggestion
```